### PR TITLE
feat: clickup-#3qftyd | Prateek | Stage | UI - Implement new voucher logic

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
https://app.clickup.com/t/3qftyd

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

**Review by aditi-tiwari17 — APPROVED** [review]

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts [hunk 1]
```diff
@@ -643,4 +702,116 @@ export class AdvanceReceiptReportComponent implements AfterViewInit, OnDestroy,
...
+    }
+
+    /**
+     * This will preview voucher
+     *
+     * @param {*} receipt
+     * @memberof AdvanceReceiptReportComponent
+     */
+    public previewVoucher(receipt: any): void {
+        // if (this.voucherApiVersion === 2) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts:723]: ⚠️ Please check method has commented code in it

File: apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts [hunk 2]
```diff
@@ -0,0 +1,762 @@
...
+    }
+
+    /**
+     * This will preview voucher
+     *
+     * @param {*} payment
+     * @memberof PaymentReportComponent
+     */
+    public previewVoucher(payment: any): void {
+        // if (this.voucherApiVersion === 2) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts:668]: ⚠️ Empty method definition

File: apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts (hunk 1 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/reports/components/advance-receipt-report/advance-receipt-report.component.ts:723]: yes we are not allowing preview/edit currently.

**Review by ravinderwalkover — APPROVED** [review]

File: apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts (hunk 2 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/reports/components/payment-report/payment-report.component.ts:668]: yes we are not allowing preview/edit currently.

File: apps/web-giddh/src/app/payment-receipt/components/create-edit/payment-receipt.component.ts [hunk 3]
```diff
@@ -0,0 +1,1824 @@
...
+     * Initializes the component
+     *
+     * @memberof PaymentReceiptComponent
+     */
+    public ngOnInit(): void {
+        if (this.generalService.voucherApiVersion === 1) {
+            this.router.navigate(['pages', 'home']);
+        }
+
+        this.imgPath = (isElectron || isCordova) ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
```
[the remark this answers is not in the document]
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/payment-receipt/components/create-edit/payment-receipt.component.ts:372]: Done

**Pull request merged**